en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,259 @@
+
+To check if the serial port is available, in the RPi's terminal, confirm the list of serial devices by typing:
+
+```
+cd /
+ls /dev/ttyAMA0
+```
+
+On RPi, the default name of a RX/TX connection is `/dev/ttyAMA0`. The same serial port is also available as `/dev/serial0`.
+
+Next, install the mavproxy in the RPi's terminal:
+
+```
+sudo apt install python3-pip
+sudo pip3 install mavproxy
+sudo apt remove modemmanager
+```
+
+## MAVlink communication
+
+Connect the Pixhawk with the laptop via `USB` cable and change the following parameters in QGroundControl:
+
+```
+MAV_0_CONFIG = TELEM2
+XRCE_DDS_0_CFG = Disabled
+SER_TEL1_BAUD = 57600
+```
+
+Next, connect the Pixhawk `TELEM2` pins TX/RX/Ground on RPi correspondent pins. We are going to leave the `TELEM1` port for a setup with a Radio Controller.
+
+The connections listed below must be done:
+
+1. `TELEM2` TX -> RPi RXD (GPIO 15 - pin 10)
+2. `TELEM2` RX -> RPi TXD (GPIO 14 - pin 8)
+3. `TELEM2` GND -> RPi Ground (pin 6)
+
+It is important to identify the correct pins in the hardware. The image illustrates the `TELEM2` pins in the Pixhawk-6C Model.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pixhawk_pins.png" width="914" height="360" /> 
+
+Ps.: The pins are numbered according to the following:
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pins_numbers.png" width="615" height="232" /> 
+
+The following image illustrates the correspondent pins at the RPi GPIO.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/rpi_gpio.png" width="845" height="457" /> 
+
+To supply the Pixhawk with 5 Volts is possible to follow with the connection of `POWER1` port with a LiPO battery, via Power Module (PM02, for example). Another option, for this setup moment, is to supply the 5 Volts via the connection of the Pixhawk's `USB` with a smartphone charger, for example, or by keeping it in the laptop `USB`. 
+
+Also, for this setup moment, the RPi is charged with 5 Volts via Power Supply (external source/charger). 
+
+Ps.: For an onboard setup, in the drone, it is not recommended to connect the `TELEM2` VCC to supply the RPi in the 5V power (pin 4). You can use your own power supply board to power the RPi baseboard. 
+
+Next, run the mavproxy connection at `/dev/ttyAMA0`. In the RPi's terminal:
+
+```
+sudo mavproxy.py --master=/dev/serial0 --baudrate 57600
+```
+
+At this moment, you are going to see in the RPi's terminal the MAVlink communication established with the Pixhawk, via RX/TX pins. 
+The MAVlink communication is also possible via USB connection, by changing the command to:
+
+```
+sudo chmod a+rw /dev/ttyACM0
+sudo mavproxy.py --master=/dev/ttyACM0 --baudrate 57600
+```
+
+The steps above are important to double check that the hardware communication is done. In the sequence, we are going to move to the XRCE_DDS communication.
+
+## ROS setup on RPi
+
+To install ROS 2 Humble you can follow the official tutorial, available in the website:
```

Comment:
I will probably replace this with link to http://docs.px4.io/main/en/ros/ros2_comm.html (or at least sections in it).

Reply:
If you see the section "Install ROS 2", 

http://docs.px4.io/main/en/ros/ros2_comm.html#install-ros-2

there is a link to:

https://docs.ros.org/en/foxy/Installation/Ubuntu-Install-Debians.html.

I only change it directly to the Humble link, instead of Foxy.

Reply:
I will probably refer to both.